src/plugins/dashboard/public/dashboard_listing/hooks/use_dashboard_listing_table.tsx

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback, useState, useMemo } from 'react';
+import type { SavedObjectsFindOptionsReference } from '@kbn/core/public';
+import { reportPerformanceMetricEvent } from '@kbn/ebt-tools';
+import {
+  TableListViewTableProps,
+  type UserContentCommonSchema,
+} from '@kbn/content-management-table-list-view-table';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+
+import { DashboardListingEmptyPrompt } from '../dashboard_listing_empty_prompt';
+import { pluginServices } from '../../services/plugin_services';
+import {
+  DASHBOARD_CONTENT_ID,
+  SAVED_OBJECT_DELETE_TIME,
+  SAVED_OBJECT_LOADED_TIME,
+} from '../../dashboard_constants';
+import { DashboardItem } from '../../../common/content_management';
+import {
+  dashboardListingErrorStrings,
+  dashboardListingTableStrings,
+} from '../_dashboard_listing_strings';
+import { confirmCreateWithUnsaved } from '../confirm_overlays';
+
+interface DashboardSavedObjectUserContent extends UserContentCommonSchema {
+  attributes: {
+    title: string;
+    description?: string;
+    timeRestore: boolean;
+  };
+}
+
+type GetDetailViewLink =
+  TableListViewTableProps<DashboardSavedObjectUserContent>['getDetailViewLink'];
+
+const SAVED_OBJECTS_LIMIT_SETTING = 'savedObjects:listingLimit';
+const SAVED_OBJECTS_PER_PAGE_SETTING = 'savedObjects:perPage';
+
+const toTableListViewSavedObject = (hit: DashboardItem): DashboardSavedObjectUserContent => {
+  const { title, description, timeRestore } = hit.attributes;
+  return {
+    type: 'dashboard',
+    id: hit.id,
+    updatedAt: hit.updatedAt!,
+    references: hit.references,
+    attributes: {
+      title,
+      description,
+      timeRestore,
+    },
+  };
+};
+
+export const useDashboardListingTable = ({
+  dashboardListingId = 'dashboard',
+  disableCreateDashboardButton,
+  getDashboardUrl,
+  goToDashboard,
+  headingId = 'dashboardListingHeading',
+  initialFilter,
+  urlStateEnabled,
+  useSessionStorageIntegration,
+}: {
+  dashboardListingId?: string;
+  disableCreateDashboardButton?: boolean;
+  getDashboardUrl: (dashboardId: string, usesTimeRestore: boolean) => string;
+  goToDashboard: (dashboardId?: string, viewMode?: ViewMode) => void;
+  headingId?: string;
+  initialFilter?: string;
+  urlStateEnabled?: boolean;
+  useSessionStorageIntegration?: boolean;
+}) => {
```

Comment:
Do you think we can be explicit about the type the hook is expected to return here?